scripts/unminimize.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+# A script allowing to minimize and un-minimize clients in a LIFO way
+# (last minimized client will be un-minimized first).
+# 
+# `chmod +x unminimize.sh` then call `./unminimize.sh`.
+
+
+Mod=${Mod:-Mod4}
+Minimizekey=Shift-m
+Unminimizekey=Ctrl-m
+SCRIPT_PATH="$HOME/.config/herbstluftwm/unminimize.sh"
```

Comment:
I ended up with 
`SCRIPT_PATH="$(cd "$(dirname "$0")" ; pwd)/$(basename "$0")"`
which is a bit convoluted, but seems rather robust.

Reply:
This does not work if the script is in the `$PATH`. Also, is the nesting of double-quotes really sane? I think, you can even omit the outer quotes, because `SCRIPT_PATH=$(...)` already works. (You later use `$SCRIPT_PATH` without quotes, so spaces will break there)

Reply:
> This does not work if the script is in the `$PATH`.

`$0`, in this case, returns the absolute path (I've just tested!), so it seems good to me. Am i wrong?

> Also, is the nesting of double-quotes really sane? I think, you can even omit the outer quotes, because `SCRIPT_PATH=$(...)` already works

Hmm... We need to quote:
* `"$(dirname "$0")" ` because of `cd`
* `"$0"`

So I can use:

`SCRIPT_PATH=$(cd "$(dirname "$0")" ; pwd)/$(basename "$0")`

and 

`hc keybind $Mod-$Minimizekey spawn "$SCRIPT_PATH" minimize`


Nesting of double quotes is possible here due to `$(...)`.
See: https://unix.stackexchange.com/questions/289574/nested-double-quotes-in-highly-voted-one-liner
(I've just found this link and does not had time yet to investigate the use of `BASH_SOURCE`)

Reply:
> > This does not work if the script is in the `$PATH`.
> 
> `$0`, in this case, returns the absolute path (I've just tested!), so it seems good to me. Am i wrong?

Ah, indeed. I though this was different, but apparently I was wrong. Thanks for checking!

> > Also, is the nesting of double-quotes really sane? I think, you can even omit the outer quotes, because `SCRIPT_PATH=$(...)` already works
> 
> Hmm... We need to quote:
> 
> * `"$(dirname "$0")" ` because of `cd`
> * `"$0"`
> 
> So I can use:
> 
> `SCRIPT_PATH=$(cd "$(dirname "$0")" ; pwd)/$(basename "$0")`

Am I right that you didn't want `SCRIPT_PATH=$0` because `$0` could be a relative path? I first thought one could use `realpath -s "$0"` instead, but `realpath` does seem to be [quite a recent utility in coreutils](https://unix.stackexchange.com/questions/101080/realpath-command-not-found)

> `hc keybind $Mod-$Minimizekey spawn "$SCRIPT_PATH" minimize`
> 
> Nesting of double quotes is possible here due to `$(...)`.
> See: https://unix.stackexchange.com/questions/289574/nested-double-quotes-in-highly-voted-one-liner
> (I've just found this link and does not had time yet to investigate the use of `BASH_SOURCE`)

Ah, I see, thanks!

Reply:
> Am I right that you didn't want `SCRIPT_PATH=$0` because `$0` could be a relative path?

yes, exactly.

> I first thought one could use `realpath -s "$0"` instead, but `realpath` does seem to be [quite a recent utility in coreutils](https://unix.stackexchange.com/questions/101080/realpath-command-not-found)

In fact, I tested this command, which works perfectly well, but I was reticent to include it for the very same reason.

> 
> > `hc keybind $Mod-$Minimizekey spawn "$SCRIPT_PATH" minimize`
> > Nesting of double quotes is possible here due to `$(...)`.
> > See: https://unix.stackexchange.com/questions/289574/nested-double-quotes-in-highly-voted-one-liner
> > (I've just found this link and does not had time yet to investigate the use of `BASH_SOURCE`)
> 
> Ah, I see, thanks!

Eventually, BASH_SOURCE allow to handle the `source` case (`. ./unminimize.sh`), so I updated the command based on the link above:

`SCRIPT_PATH=$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )/$(basename "${BASH_SOURCE[0]}")`

For the record, the `>` redirection here allows the command to work even if a user has defined an alias on `cd` which returns some result (the last directory before the change, for instance).